OpenRA.Mods.Common/UtilityCommands/ExtractWeaponDocsCommand.cs

```diff
@@ -36,66 +37,54 @@ void IUtilityCommand.Run(Utility utility, string[] args)
-			foreach (var t in weaponTypes)
-			{
-				// skip helpers like TraitInfo<T>
-				if (t.ContainsGenericParameters || t.IsAbstract)
-					continue;
-
-				if (currentNamespace != t.Namespace)
-				{
-					currentNamespace = t.Namespace;
-					doc.AppendLine();
-					doc.AppendLine($"## {currentNamespace}");
-				}
-
-				var traitName = t.Name.EndsWith("Info") ? t.Name.Substring(0, t.Name.Length - 4) : t.Name;
-				doc.AppendLine();
-				doc.AppendLine($"### {traitName}");
+			var weaponTypes = weaponInfo.Concat(projectiles).Concat(warheads);
 
-				var traitDescLines = t.GetCustomAttributes<DescAttribute>(false).SelectMany(d => d.Lines);
-				foreach (var line in traitDescLines)
-					doc.AppendLine(line);
-
-				var infos = FieldLoader.GetTypeLoadInfo(t);
-				if (!infos.Any())
-					continue;
-
-				doc.AppendLine();
-				doc.AppendLine("| Property | Default Value | Type | Description |");
-				doc.AppendLine("| -------- | --------------| ---- | ----------- |");
+			var json = GenerateJson(version, weaponTypes, objectCreator);
+			Console.WriteLine(json);
+		}
 
-				var liveTraitInfo = t == typeof(WeaponInfo) ? null : objectCreator.CreateBasic(t);
-				foreach (var info in infos)
+		static string GenerateJson(string version, IEnumerable<Type> weaponTypes, ObjectCreator objectCreator)
+		{
+			var weaponTypesInfo = weaponTypes.Where(x => !x.ContainsGenericParameters && !x.IsAbstract)
+				.Select(t => new
+				{
+					Type = t,
+					Name = t.Name.EndsWith("Info") ? t.Name.Substring(0, t.Name.Length - 4) : t.Name
+				})
+				.Select(x => new
 				{
-					var fieldDescLines = info.Field.GetCustomAttributes<DescAttribute>(true).SelectMany(d => d.Lines);
-					var fieldType = Util.FriendlyTypeName(info.Field.FieldType);
-					var defaultValue = liveTraitInfo == null ? "" : FieldSaver.SaveField(liveTraitInfo, info.Field.Name).Value.Value;
-					doc.Append($"| {info.YamlName} | {defaultValue} | {fieldType} | ");
-					foreach (var line in fieldDescLines)
-						doc.Append(line + " ");
-					doc.AppendLine("|");
-				}
-			}
+					Namespace = x.Type.Namespace,
+					Name = x.Name,
+					Description = string.Join(" ", x.Type.GetCustomAttributes<DescAttribute>(false).SelectMany(d => d.Lines)),
+					InheritedTypes = x.Type.BaseTypes().Select(y => y.Name).Where(y => y != x.Name && y != $"{x.Name}Info" && y != "Object").ToArray(),
+					Properties = FieldLoader.GetTypeLoadInfo(x.Type).Select(fi => new
+					{
+						Name = fi.YamlName,
+						DefaultValue = FieldSaver.SaveField(objectCreator.CreateBasic(x.Type), fi.Field.Name).Value.Value,
+						Type = Util.FriendlyTypeName(fi.Field.FieldType),
+						Description = string.Join(" ", fi.Field.GetCustomAttributes<DescAttribute>(true).SelectMany(d => d.Lines)),
+						OtherAttributes = fi.Field.CustomAttributes
+							.Where(a => a.AttributeType.Name != "DescAttribute")
+							.Select(a => new
+							{
+								Name = a.AttributeType.Name.Substring(0, a.AttributeType.Name.Length - 9),
```

Comment:
Consider an `EndsWith` guard here in case there's an attribute not following the convention of using an `Attribute` suffix.